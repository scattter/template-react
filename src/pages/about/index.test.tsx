import { fireEvent, render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { About } from './index';

describe('test app about page', () => {
  it('should success render', function () {
    const history = createMemoryHistory({ initialEntries: ['/about'] });
    const wrapper = render(
      <Router location={history.location} navigator={history}>
        <About />
      </Router>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should success back to home page use history.push', function () {
    const history = createMemoryHistory({ initialEntries: ['/about'] });
    const { getByText } = render(
      <Router location={history.location} navigator={history}>
        <About />
      </Router>,
    );
    expect(history.location.pathname).toBe('/about');
    fireEvent.click(getByText('back home page with navigate(history.push) jump'));
    expect(history.location.pathname).toBe('/home');
  });

  it('should success back to home page use history.replace', function () {
    const history = createMemoryHistory({ initialEntries: ['/about'] });
    const { getByText } = render(
      <Router location={history.location} navigator={history}>
        <About />
      </Router>,
    );
    expect(history.location.pathname).toBe('/about');
    fireEvent.click(getByText('back home page with navigate(history.replace) jump'));
    expect(history.location.pathname).toBe('/home');
  });
});
