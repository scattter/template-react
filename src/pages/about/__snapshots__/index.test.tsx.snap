// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app about page should success render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="center"
      >
        <span>
          this is about page
        </span>
        <br />
        <button>
          back home page with navigate(history.back) back
        </button>
        <br />
        <button>
          back home page with navigate(history.push) jump
        </button>
        <br />
        <button>
          back home page with navigate(history.replace) jump
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="center"
    >
      <span>
        this is about page
      </span>
      <br />
      <button>
        back home page with navigate(history.back) back
      </button>
      <br />
      <button>
        back home page with navigate(history.push) jump
      </button>
      <br />
      <button>
        back home page with navigate(history.replace) jump
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
